code/go/0chain.net/smartcontract/minersc/test_helper.go

```diff
@@ -96,105 +95,127 @@ func newClient(balance state.Balance, balances cstate.StateContextI) (
-	var (
-		input = c.addNodeRequest(t, delegateWallet)
-		gn    *GlobalNode
-	)
-	gn, err = msc.getGlobalNode(balances)
-	require.NoError(t, err, "missing global node")
-	return msc.AddMiner(tx, input, gn, balances)
+type TestClient struct {
+	client   *Client
+	delegate *Client
+	stakers  []*Client
 }
 
-func (c *Client) callAddSharder(t *testing.T, msc *MinerSmartContract,
-	now int64, delegateWallet string, balances cstate.StateContextI) (
-	resp string, err error) {
+func createLFMB(miners []*TestClient, sharders []*TestClient) (
+	b *block.Block) {
 
-	var tx = newTransaction(c.id, ADDRESS, 0, now)
-	balances.(*testBalances).txn = tx
-	var (
-		input = c.addNodeRequest(t, delegateWallet)
-		gn    *GlobalNode
-	)
-	gn, err = msc.getGlobalNode(balances)
-	require.NoError(t, err, "missing global node")
-	return msc.AddSharder(tx, input, gn, balances)
-}
+	b = new(block.Block)
 
-func addMiner(t *testing.T, msc *MinerSmartContract, now int64,
-	balances cstate.StateContextI) (miner, delegate *Client) {
+	b.MagicBlock = block.NewMagicBlock()
+	b.MagicBlock.Miners = node.NewPool(node.NodeTypeMiner)
+	b.MagicBlock.Sharders = node.NewPool(node.NodeTypeSharder)
 
-	miner, delegate = newClient(0, balances), newClient(0, balances)
-	var err error
-	_, err = miner.callAddMiner(t, msc, now, delegate.id, balances)
-	require.NoError(t, err, "add_miner")
+	for _, miner := range miners {
+		b.MagicBlock.Miners.NodesMap[miner.client.id] = new(node.Node)
+	}
+	for _, sharder := range sharders {
+		b.MagicBlock.Sharders.NodesMap[sharder.client.id] = new(node.Node)
+	}
 	return
 }
 
-func addSharder(t *testing.T, msc *MinerSmartContract, now int64,
-	balances cstate.StateContextI) (miner, delegate *Client) {
+// create and add miner/sharder, create stake holders, don't stake
+func newClientWithStakers(isMiner bool, t *testing.T, msc *MinerSmartContract,
+	now, stakersAmount int64, stakeValue state.Balance,
+	balances cstate.StateContextI) (
+		client *TestClient) {
 
-	miner, delegate = newClient(0, balances), newClient(0, balances)
-	var err error
-	_, err = miner.callAddSharder(t, msc, now, delegate.id, balances)
-	require.NoError(t, err, "add_sharder")
+	client = new(TestClient)
+	client.client, client.delegate = newClientWithDelegate(isMiner, t, msc, now, balances)
+	for i := int64(0); i < stakersAmount; i++ {
+		client.stakers = append(client.stakers, newClient(stakeValue, balances))
+	}
 	return
 }
 
-func (c *Client) addToDelegatePoolRequest(t *testing.T, nodeID string) []byte {
-	var dp deletePool
-	dp.MinerID = nodeID
-	return mustEncode(t, &dp)
-}
-
 // stake a miner or a sharder
 func (c *Client) callAddToDelegatePool(t *testing.T, msc *MinerSmartContract,
-	now, val int64, nodeID string, balances cstate.StateContextI) (resp string,
-	err error) {
+	now int64, value state.Balance, nodeId string,
+	balances cstate.StateContextI) (resp string, err error) {
 
 	t.Helper()
-	var tx = newTransaction(c.id, ADDRESS, val, now)
+
```

Comment:
`var` usage kind of messy. Probably merge three var statements, or split into six!

Reply:
Fixed this and one more below, let me know if still can be improved